cf/regrid/regrid.py

```diff
@@ -86,6 +92,24 @@ class Grid:
     # for a 2-d grid for which one of the dimensions is a size 2 dummy
     # dimension.
     dummy_size_2_dimension: bool = False
+    # Whether or not the grid is a structured grid.
+    is_grid: bool = False
+    # Whether or not the grid is a UGRID mesh.
+    is_mesh: bool = False
```

Comment:
A brief indicator how these (`is_grid` and `is_mesh` flags) relate to each other would be helpful here, since (as I refer to in a later comment) it seems to me like we set them as complementary (i.e. `is_mesh = not is_grid` and vice versa) in the code so far added here, and contextually it seems like they should be complementary as to distinguishing unstructured from structured, but then I wonder why you've created two separate variables. Can we consolidate to use just one instead of this pair? (Maybe I am missing something here.)

Reply:
Yes - they are currently complementary, but that may change in the future, e.g. when we support HEALPix grids, which are neither structured grids nor meshes. So it's probably best to not restrict things further in the comments.